src/node-jspm-jasmine.js

```diff
@@ -141,17 +180,73 @@ export function runTests(opts, errCallback = function() {}) {
 							importTheseTestFiles();
 						}
 					})
-					.catch(errCallback);
+					.catch((ex) => {
+						if (opts.coverage && opts.coverage.tempDirectory) {
+							// remove temporary directory
+							rimraf.sync(opts.coverage.tempDirectory)
+						}
+						errCallback(ex)
+					});
 				});
 			});
 		}
 	} catch(ex) {
-		removeTempFiles();
+		if (opts.coverage && opts.coverage.tempDirectory) {
+			// remove temporary directory
+			rimraf.sync(opts.coverage.tempDirectory)
+		}
 		errCallback(new Error(`Jasmine or Jspm may not be properly configured -- '${ex.toString()}'`));
 	}
 }
 
 function importTestFiles(SystemJS, jasmine, specDir, specFiles, coverage, errCallback) {
+
+	if (coverage) {
+		// the "onComplete" hook to prevent jasmine's self-righteous exit
+		jasmine.onComplete(function (passed) {
+			// avoid misalignment with jasmine's output
+			console.log('');
+			// import the rest of the modules not already imported (evaluated)
+			// as dependencies of specs
+			Promise.all(Object.keys(coverage.coverageFiles).map(function (file) {
+				return SystemJS.import(path.join(process.cwd(), file));
+			})).then(function () {
+				if (typeof __coverage__ === 'undefined') {
+					console.log("No coverage was collected for files matching globs " + coverage.files);
+				} else {
+					try {
+						const collector = remap(__coverage__)
+						let report = Report.create(
+							(coverage.reporter || 'html'),
+							{
+								dir: coverage.dir
+							}
+						);
+						report.writeReport(collector, true);
```

Comment:
@joeldenning Your comment is a bit confusing to me, because as it stands in the latest pr, there is no dying without error (if that's not what you're experiencing, please elaborate) -- sourcemap or not, it's getting instrumented and included in the coverage report. And I don't have a problem with that so as long as we explain in the README what it means to have a directory in your coverage report called "no-source-map". Alternatively we could let it do its thing with the "no-source-map" and also issue a warning explaining why that happened. Or another option is just to refuse to process files that were suppose to have had a sourcemap and throw the exception accordingly (whether that exception is a warning or fatal error is up to you). Any of those three options is reasonable to me as it causes no debugging issues, it's more of user experience detail.